x/superfluid/keeper/msg_server.go

```diff
@@ -46,6 +46,8 @@ func (server msgServer) SuperfluidDelegate(goCtx context.Context, msg *types.Msg
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
 	err := server.keeper.SuperfluidDelegate(ctx, msg.Sender, msg.LockId, msg.ValAddr)
+
+	// NOTE: `EmitDelegationEvent` (x/staking module's delegation event) happens in keeper level
 	if err == nil {
 		events.EmitSuperfluidDelegateEvent(ctx, msg.LockId, msg.ValAddr)
 	}
```

Comment:
Im really confused by this entire msg_server. Why don't we just treat err as err != nil and return early rather than doing some logic off of a nil error? Im fairly certain all the other msg servers are like that.

As a side note, if we are going to emit the staking module delegation event at the keeper level but not the user delegation event, we should either

1. Add a comment explaining why there is this separation or
2. Also emit the user delegation event at the keeper level and modify the comment to say that.

Reply:
Addressed here, https://github.com/osmosis-labs/osmosis/pull/8286/commits/7364ee284e88372852fd2c3029f52ef554fd41ed let me know if this should suffice!